includes/discovery/discovery-protocols.inc.php

```diff
@@ -81,7 +81,45 @@
     echo PHP_EOL;
 }//end if
 
-if (($device['os'] == 'pbn' || $device['os'] == 'bdcom') && Config::get('autodiscovery.xdp') === true) {
+if (($device['os'] == 'routeros') && Config::get('autodiscovery.xdp') === true) {
+    echo ' LLDP-MIB: ';
+    $lldp_array  = snmpwalk_group($device, 'lldpRemEntry', 'LLDP-MIB', 3);
+    $lldp_ports = snmpwalk_group($device, 'mtxrInterfaceStatsName', 'MIKROTIK-MIB');
```

Comment:
Do not unconditionally do 3 snmpwalks here. 

If the first one fails, there is no point in doing the other two.